component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/reflect/JavaVersion.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (C) 2006-2022 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.reflect;
+
+public class JavaVersion {
+
+    /**
+     * Retrieve major version
+     *
+     * @param version
+     * @return major version
+     */
+    public static String getMajor(final String version) {
```

Comment:
advice make all methods to "private" if possible, and if for test, please remove the "public" to use default.

And LGTM, and not sure if we need so many methods here.

Reply:
Effectively, the need is only major but which can most can least ;-) May be useful in some cases some day...